contracts/pcv/morpho/MorphoCompoundPCVDeposit.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {ILens} from "./ILens.sol";
+import {IMorpho} from "./IMorpho.sol";
+import {CoreRef} from "../../refs/CoreRef.sol";
+import {PCVDeposit} from "../PCVDeposit.sol";
+import {ICompoundOracle, ICToken} from "./ICompound.sol";
+
+/// @notice PCV Deposit for Morpho-Compound V2.
+/// Implements the PCV Deposit interface to deposit and withdraw funds in Morpho
+/// Liquidity profile of Morpho for this deposit is fully liquid for USDC and DAI
+/// because the incentivized rates are higher than the P2P rate.
+/// Only for depositing USDC and DAI. USDT is not in scope.
+/// @dev approves the Morpho Deposit to spend this PCV deposit's token,
+/// and then calls supply on Morpho, which pulls the underlying token to Morpho,
+/// drawing down on the approved amount to be spent,
+/// and then giving this PCV Deposit credits on Morpho in exchange for the underlying
+/// @dev PCV Guardian functions withdrawERC20ToSafeAddress and withdrawAllERC20ToSafeAddress
+/// will not work with removing Morpho Tokens on the Morpho PCV Deposit because Morpho
+/// has no concept of mTokens. This means if the contract is paused, or an issue is
+/// surfaced in Morpho and liquidity is locked, Volt will need to rely on social
+/// coordination with the Morpho team to recover funds.
+contract MorphoCompoundPCVDeposit is PCVDeposit {
+    using SafeERC20 for IERC20;
+
+    /// @notice reference to the lens contract for morpho-compound v2
+    address public constant LENS = 0x930f1b46e1D081Ec1524efD95752bE3eCe51EF67;
+
+    /// @notice reference to the morpho-compound v2 market
+    IMorpho public constant MORPHO =
+        IMorpho(0x8888882f8f843896699869179fB6E4f7e3B58888);
+
+    /// @notice reference to underlying token
+    address public immutable token;
+
+    /// @notice cToken in compound this deposit tracks
+    /// used to inform morpho about the desired market to supply liquidity
+    address public immutable cToken;
+
+    constructor(
+        address _core,
+        address _cToken,
+        address _underlying
+    ) CoreRef(_core) {
+        cToken = _cToken;
+        token = _underlying;
```

Comment:
could check that cToken.underlying() == token but I don't feel strongly about it

Reply:
If we check, that breaks compatibility with ceth, unless we surround the check with a try catch that only checks if the call succeeds.